compiler/modules/front/codeGen.bal

```diff
@@ -917,6 +918,7 @@ function codeGenAssignToMember(CodeGenContext cx, bir:BasicBlock startBlock, Env
 }
 
 function codeGenCompoundAssignStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environment env, s:CompoundAssignStmt stmt) returns CodeGenError|StmtEffect {
+    cx.skipNilLiftig = true;
```

Comment:
This is not OK. It looks like you are using this to deal with not doing nil lifting in compound assignment.

The way to handle this is to have a clean layering between the functions that do nil lifting and the functions that do the underlying operation, and then use the latter when doing compound assignment.

You may also need to refactor the compound assignment to not use codeGenExpr for the operator in the compound assignement, but instead use a lower-level function.

Reply:
fixed.